formatter/src/formatter/mac.rs

```diff
@@ -155,16 +155,29 @@ pub fn format_macro(
     settings: &FormatterSettings,
     source: Option<&Rope>,
 ) -> String {
-    let mut printer = Printer::new(settings.into());
+    let mut printer: Printer;
     let mut formatter = match source {
         Some(source) => {
             let whitespace = crate::collect_comments::extract_whitespace_and_comments(
                 source,
                 mac.mac.tokens.clone(),
             );
+            let crlf_line_endings = source
+                .raw_lines()
+                .nth(0)
+                .unwrap()
+                .to_string()
+                .ends_with("\r\n");
```

Comment:
Let's avoid panicking if the file is empty
```suggestion
            let crlf_line_endings = source
                .raw_lines()
                .nth(0)
                .map(|raw_line| raw_line.to_string().ends_with("\r\n"))
                .unwrap_or_default();
```

Reply:
Done